packages/babel-helper-create-class-features-plugin/src/decorators.ts

```diff
@@ -1493,36 +1677,86 @@ function transformClass(
       superClass = id;
     }
   }
-  originalClass.body.body.unshift(
-    t.staticBlock(
-      [
+
+  const applyDecoratorWrapper = t.staticBlock([]);
+  originalClass.body.body.unshift(applyDecoratorWrapper);
+  const applyDecsBody = applyDecoratorWrapper.body;
+  if (computedKeyAssignments.length > 0) {
+    const elements = originalClassPath.get("body.body");
+    let lastPublicElement: NodePath<t.ClassProperty | t.ClassMethod>;
```

Comment:
The `computedKeyAssignments` hosts decorator and computed keys evaluation results. They will be referenced in the `applyDecs` call emited in the first static block of the class. I think the output will be slightly more readable if we insert the assignments to the _first_ public element, so that they will not be too far away from where they will be used.

Reply:
https://github.com/babel/babel/pull/16325/commits/e856f3ccae87fbf5da14cdf41c12de34d6ef6910